File: fixtures/scale/haskell/large/Ciphers/Trivium80/Impl.hs
module Ciphers.Trivium80.Impl (encrypt, decrypt) where

import Ciphers.Trivium80.Model (keyBits, rounds)

encrypt :: Int -> [Int] -> [Int]
encrypt key = map (\b -> mod (b + key + rounds + keyBits) 256)

decrypt :: Int -> [Int] -> [Int]
decrypt key = map (\b -> mod (b - key - rounds - keyBits + 512) 256)

File: fixtures/scale/elixir/large/lib/ciphers/trivium80/impl.ex
defmodule Ciphers.Trivium80.Impl do
  alias Ciphers.Trivium80.Model

  def encrypt(key, bytes) do
    Enum.map(bytes, &rem(&1 + key + Model.rounds() + Model.key_bits(), 256))
  end

  def decrypt(key, bytes) do
    Enum.map(bytes, &rem(&1 - key - Model.rounds() - Model.key_bits() + 512, 256))
  end
end
